### DISCUSSION TITLE
Please provide logs to help us troubleshoot your problems

### DISCUSSION THREAD

**Discussion by peppy** [ppy/osu#15188] [Q&A]
Please attach logs here, which are located at:

- `%AppData%/osu/logs` (on Windows),
- `~/.local/share/osu/logs` (on Linux),
- `~/Library/Application Support/osu/logs` (on macOS),
- `Android/data/sh.ppy.osulazer/files/logs` (on Android),
- on iOS they can be obtained by connecting your device to your desktop and copying the `logs` directory from the app's own document storage using iTunes. (https://support.apple.com/en-us/HT201301#copy-to-computer)

You can drag them into the github message composer to upload them alongside your discussion post.

**Comment by bdach** [comment]
Sometimes you may be prompted by us to provide Windows Event Viewer logs as a follow up. This can be done by performing the following steps:
1. Type `Event viewer` in the Windows search bar (shortcut: <kbd>Windows</kbd>+<kbd>S</kbd>) and press <kbd>Enter</kbd>, or open the *Run...* button (<kbd>Windows</kbd>+<kbd>R</kbd>), type in `eventvwr` and press <kbd>Enter</kbd>.
2. The *Event viewer* window will appear. Expand *Windows Logs* on the left and choose *Application*. It might take a while to load.
3. In the *Actions* pane on the right click *Save All Events As...*, then choose a location to save to.
4. On the *Display Information* popup, select the second option, then the checkbox to show all available languages, then *English (UK)* (US would also do fine).
5. Append the resulting `.evtx` file to the relevant issue or forum thread.
